src/GrpcProtos/capture.proto

```diff
@@ -97,12 +97,10 @@ message InternedCallstack {
 }
 
 message CallstackSample {
+  reserved 3, 4;
   int32 pid = 1;
   int32 tid = 2;
-  oneof callstack_or_key {
-    Callstack callstack = 3;
-    uint64 callstack_key = 4;
-  }
+  uint64 callstack_key = 6;
```

Comment:
I'm against getting rid of this `oneof` and I don't see a reason for it. The change of signature of `OnCallstackSample` shows how this just causes a hack now: the `CallstackSample` proto is incomplete when sent and the `key` has to be added afterwards. At the same time, the actual callstack has to be sent as a separate vector parameter, which is inconsistent with the rest of the methods in `TracerListener`.

Reply:
Ok, here is the thing - because of this one-of one of the client tests was testing non-existing scenario. I think we shouldn't keep fields in proto we do not use for communication purposes. having "simple" interface of TraceListener is not good enough reason. we either move Intern* processing up one level or change the interface.

Reply:
Ideally, `TracerListener` wouldn't receive protos but normal struct and it would translate to protos. I'm aware of this but it's quite an additional maintenance work. That's why we did it like this. Your proposed solution is a middle ground that ends up being even more confusing: it's not a matter of "simple", it's a matter of "consistent", which is different.
You say that the interface of `TracerListener` is not a good enough reason, but you don't give a reason to remove the field, other than "we shouldn't keep it". It serves our purpose well, it obviously doesn't take actual space and we can test/CHECK appropriately where we don't expect it to be set. And it's conceptually correct: the event can contain the callstack or the key to an interned callstack. If a user of the proto wants to take out the callstack and intern it, it can do it, which is exactly what happens now.

Reply:
after discussing offline we decided to move this part to follow-up PR